### PR TITLE
0009626: Added reloadacl command for server console

### PR DESCRIPTION
**Pull request by Timic3**
This simple pull request adds reloadacl server command. See [#9626](https://bugs.multitheftauto.com/view.php?id=9626)

### CODE REVIEW

**Comment by qaisjp** [comment]
Hey @Timic3! Thank you for your pull request.

The code looks great (although I am concerned about whether only administrators can run the command...), but I think it would be best to actually add this command to [the `admin` resource](https://github.com/multitheftauto/mtasa-resources/tree/master/%5Badmin%5D/admin).

Would you like to work on that? I'm happy to assign [mantis 9626](https://bugs.multitheftauto.com/view.php?id=9626) to you!

**Comment by Timic3** [comment]
Hi @qaisjp, the main reason I added this is because when people start the server for the first time, they don't edit the ACL (at least I don't) and to add themselves to Admin ACL group, they'd need to restart the server. If this would be added to the admin resource, will console still be able to use /reloadacl? And if people didn't download the resources package, they wouldn't be able to use this command.

I think players cannot use this command (correct me if I am wrong), because I did this only for server console.

**Comment by Dezash** [comment]
Console has Admin permissions by default so they would be able to use it.

**Comment by Timic3** [comment]
Still, that doesn't eliminate the problem where they haven't downloaded the resources.

**Comment by 4O4** [comment]
I agree with @Timic3. ACL is built-in MTA feature, which is tightly coupled with the xml config file. IMO there should also be a built-in command for such trivial task as reloading this config, which eliminates the need to install / code by hand any additional resources.

**Comment by qaisjp** [comment]
These are reasonable arguments. 

Also, to answer your question, the console can execute normal `addCommandHandler` commands.

Thanks again for your pull request!

**Review by qaisjp — APPROVED** [review]

**Comment by ccw808** [comment]
I think the command needs to block anyone from using it. See the example at the start of CConsoleCommands::AuthSerial

**Comment by qaisjp** [comment]
I'm not sure... the `start` command doesn't seem to have the block. [See here.](https://github.com/Timic3/mtasa-blue/blob/7bf2d93793053bbdf8927ed1fb37ee1c6a161d83/Server/mods/deathmatch/logic/CConsoleCommands.cpp#L53)

**Comment by ccw808** [comment]
start command is protected in acl.xml

File: Server/mods/deathmatch/logic/CConsoleCommands.cpp [hunk 1]
```diff
@@ -1827,6 +1827,17 @@ bool CConsoleCommands::AuthorizeSerial( CConsole* pConsole, const char* szArgume
     return false;
 }
 
+bool CConsoleCommands::ReloadAcl(CConsole* pConsole, const char* szArguments, CClient* pClient, CClient* pEchoClient)
+{
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/CConsoleCommands.cpp:1831]: Needs to make sure only console / people with rights can use this command. [See this example.](https://github.com/Timic3/mtasa-blue/blob/7bf2d93793053bbdf8927ed1fb37ee1c6a161d83/Server/mods/deathmatch/logic/CConsoleCommands.cpp#L1699)

**Review by qaisjp — CHANGES_REQUESTED** [review]

**Comment by Timic3** [comment]
Okay, I added that check you requested - I also added the command to ACL, so only users in Admin group can execute it.

**Review by qaisjp — APPROVED** [review]

**Review by ccw808 — APPROVED** [review]

**Comment by Timic3** [comment]
I tested everything (reloading ACL with no rights, with rights & server console) and as far as I know, everything works. I think this can be merged, but if you want to test it again just to be sure, go ahead.

**Pull request merged**